shared/modules/module.go

```diff
@@ -20,9 +20,10 @@ type InterruptableModule interface {
 	Stop() error
 }
 
+type ModuleOption func(InitializableModule)
```

Comment:
I left a comment elsewhere, but we are passing in options to non-initilizable modules, so I'm beginning to get a bit confused.

Reply:
It's like saying:

"We support options for all modules that can be created by specifying options."

Since we are applying interface segregation, the idea is that the options should work with all modules that have `Create(bus Bus, options ...ModuleOption) (Module, error)`. 
These modules implement -at least- `InitializableModule` so we can leverage this to have a common pattern but at the same time the ability to configure specific things by just casting to the module struct. 

Does it make sense?

Reply:
Yes. I think you should add a comment explaining that `InitilizableModule` is the "minimum expected inheritance" in some sense

Reply:
Will handle in https://github.com/pokt-network/pocket/pull/520/files#r1109007712